src/heretic/utils.py

```diff
@@ -36,7 +41,35 @@ def format_duration(seconds: float) -> str:
 
 
 def load_prompts(specification: DatasetSpecification) -> list[str]:
-    dataset = load_dataset(specification.dataset, split=specification.split)
+    path = specification.dataset
+    split_str = specification.split
+    if not os.path.isdir(path):
+        # Probably a repository path; let load_dataset figure it out.
+        dataset = load_dataset(path, split=split_str)
+    elif not Path(path, DATASET_STATE_JSON_FILENAME).exists():
```

Comment:
The two consecutive `not` clauses are a bit difficult to parse. I think the following layout would be cleaner:

```python
if os.path.isdir(path):
  if Path(path, DATASET_STATE_JSON_FILENAME).exists():
    ...
  else:
    ...
else:
  ...
```